File: 169. Majority Element/169. Majority Element.py
# class Solution:
#     def majorityElement(self, nums: List[int]) -> int:
#         check = len(nums) / 2
#         arr = {}
#         result = 0
#         for i in nums:
#             if i not in arr:
#                 arr[i] = 1
#             else:
#                 arr[i] += 1
#         for i in arr:
#             if arr[i] > check:
#                 result = i
#                 break
#         return result
    
    
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 0
        number = None
        for num in nums:
            if count == 0:
                number = num
            count += 1 if num == number else -1
        return number

File: 169. Majority Element/169. Majority Element.java

// import java.util.HashMap;
// import java.util.Map;

// class Solution {

//     public int majorityElement(int[] nums) {
//         int number = 0;
//         int check = nums.length / 2;
//         Map<Integer, Integer> map = new HashMap<>();
//         for (int i : nums) {
//             map.put(i, map.getOrDefault(i, 0) + 1);
//         }
//         for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
//             if (entry.getValue() > check) {
//                 number = entry.getKey();
//             }
//         }
//         return number;
//     }
// }

class Solution {
    public int majorityElement(int[] nums) {
        int count = 0;
        int num = 0;
        for (int i : nums) {
            if (count == 0) num = i;
            if (num == i) count++;
            else count--;
        }
        return num;
    }
}